unit_testing/DC/DC_tests.cpp

```diff
@@ -0,0 +1,506 @@
+        simTick(100); /* Should respond within 100 ms */
+
+        /* Only the requested port should be on */
+        for(int j = 0; j < ACTUATIONPORTS; j++) {
+            if(j != i) {
+                ASSERT_EQ(*getTimerCCR(j), 0) << "j = " << j;    
+            } 
+            else {
+                ASSERT_EQ(*getTimerCCR(j), 999) << "j = " << j;
+            }
+        }
+
+        /* Check status bit updated correctly. Could take up to 100 ms for another print */
+        simTick(100);
+        switch(i) {
+            case 0: EXPECT_FLUSH_USB(Contains("0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0x00000001"));
+                    break;
+            case 1: EXPECT_FLUSH_USB(Contains("0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0x00000002"));
+                    break;
+            case 2: EXPECT_FLUSH_USB(Contains("0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0x00000004"));
+                    break;
+            case 3: EXPECT_FLUSH_USB(Contains("0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0x00000008"));
+                    break;
+            case 4: EXPECT_FLUSH_USB(Contains("0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0x00000010"));
+                    break;
+            case 5: EXPECT_FLUSH_USB(Contains("0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0x00000020"));
+                    break;
+            default: FAIL();
+        }
+
+        /* Release button */
+        stmSetGpio(buttonGpio[i], true);
+        simTick(100); /* Should respond within 100 ms */
+    }
+}   
+
+/* Grey box - uses getTimerCCR() */
+TEST_F(DCBoard, onboardButtonsOff) 
+{
+    dcSetup();
+    goToTick(1);
+
+    for(int i = 0; i < ACTUATIONPORTS; i++) {
+        /* All ports initially on */
+        writeDcMessage("all on\n");
+
+        for(int j = 0; j < ACTUATIONPORTS; j++) {
+            ASSERT_EQ(*getTimerCCR(j), 999) << "j = " << j;
+        }
+
+        /* Put one port on a timer */
+        char cmd[100] = {0};
+        sprintf(cmd, "p%u on %u\n", i+1, 1);
+        writeDcMessage(cmd);
+
+        /* Press a button */
+        stmSetGpio(buttonGpio[i], false);
+        simTick(100); /* Should respond within 100 ms */
+
+        /* There should be no effect */
+        for(int j = 0; j < ACTUATIONPORTS; j++) {
+            ASSERT_EQ(*getTimerCCR(j), 999) << "j = " << j;    
+        }
+
+        simTick(100); /* Could take up to 100 ms for another print */
+        /* Check status bit updated correctly */
+        EXPECT_FLUSH_USB(Contains("0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0x0000003f")); 
+
+        /* Release button */
+        stmSetGpio(buttonGpio[i], true);
+        simTick(100); /* Should respond within 100 ms */
+
+        /* Timer port should turn off 1 second after the start of the test */
+        simTick(700);
+        ASSERT_EQ(*getTimerCCR(i), 999) << "i = " << i;   
+        simTick();
+        ASSERT_EQ(*getTimerCCR(i), 0) << "i = " << i;   
+    }
+}   
+
+/* Grey box - uses getTimerCCR() */
+TEST_F(DCBoard, onboardButtonsOffDuring) 
```

Comment:
Being petty here but During what :question: -  could use a update to the test name.

Reply:
Just so you know, YOU are responsible for this:
![image](https://github.com/copenhagenatomics/CA_Embedded/assets/141138893/94f9a247-bb2e-4042-9c1b-ab1730759856)